src/TextToTalk/Backends/OpenAI/OpenAiClient.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace TextToTalk.Backends.OpenAI;
+
+public class OpenAiClient
+{
+    private const string UrlBase = "https://api.openai.com";
+
+    public static readonly IReadOnlySet<string> Models = new HashSet<string>
+    {
+        "tts-1",
+        "tts-1-hd"
+    };
+
+    public static readonly IReadOnlySet<string> Voices = new HashSet<string>
+    {
+        "alloy",
+        "echo",
+        "fable",
+        "onyx",
+        "nova",
+        "shimmer"
+    };
+
+    private readonly HttpClient client;
+    private readonly OpenAiBackendUIModel model;
+    private readonly StreamSoundQueue soundQueue;
+
+    public OpenAiClient(StreamSoundQueue soundQueue, OpenAiBackendUIModel model, HttpClient http)
+    {
+        this.soundQueue = soundQueue;
+        this.model = model;
```

Comment:
Would prefer if this took an interface/provider/something that just gives the current API key value, rather than giving it the entire UI model. You can look at what I do now for passing WSServer config options directly as an example - planning on using that pattern more often.

This will matter more if OpenAI adds an endpoint for listing voices, in which case calling that would be part of the UIModel class, and there would be a circular dependency between the two.

Reply:
Updated;
ApiKey is now a field on OpenAiClient
Removed OpenAiBackendUiModel since its no longer required